test/contracts/meta-oracles/driftlessHistoricalPriceFeed.spec.ts

```diff
@@ -0,0 +1,462 @@
+      ethPrice = ether(150);
+      await oracleWrapper.updateMedianizerPriceAsync(
+        ethMedianizer,
+        ethPrice,
+        SetTestUtils.generateTimestamp(1000),
+      );
+
+      subjectUpdateFrequency = ONE_DAY_IN_SECONDS.div(4);
+      subjectUpdateTolerance = subjectUpdateFrequency.div(4);
+      subjectMedianizerAddress = ethMedianizer.address;
+      subjectDataDescription = '200DailyETHPrice';
+      subjectSeededValues = [];
+    });
+
+    async function subject(): Promise<DriftlessHistoricalPriceFeedContract> {
+      return oracleWrapper.deployDriftlessHistoricalPriceFeedAsync(
+        subjectUpdateFrequency,
+        subjectUpdateTolerance,
+        subjectMedianizerAddress,
+        subjectDataDescription,
+        subjectSeededValues,
+      );
+    }
+
+    it('sets the correct updateFrequency', async () => {
+      historicalPriceFeed = await subject();
+
+      const actualUpdateFrequency = await historicalPriceFeed.updateFrequency.callAsync();
+
+      expect(actualUpdateFrequency).to.be.bignumber.equal(subjectUpdateFrequency);
+    });
+
+    it('sets the correct medianizer address', async () => {
+      historicalPriceFeed = await subject();
+
+      const actualMedianizerAddress = await historicalPriceFeed.medianizerInstance.callAsync();
+
+      expect(actualMedianizerAddress).to.equal(subjectMedianizerAddress);
+    });
+
+    it('sets the correct data description', async () => {
+      historicalPriceFeed = await subject();
+
+      const actualDataDescription = await historicalPriceFeed.dataDescription.callAsync();
+
+      expect(actualDataDescription).to.equal(subjectDataDescription);
+    });
+
+    it('sets the nextAvailableUpdate timestamp to the block timestamp plus 24 hours', async () => {
+      historicalPriceFeed = await subject();
+
+      const block = await web3.eth.getBlock('latest');
+      const expectedTimestamp = new BigNumber(block.timestamp).add(subjectUpdateFrequency);
+
+      const actualTimestamp = await historicalPriceFeed.nextAvailableUpdate.callAsync();
+
+      expect(actualTimestamp).to.be.bignumber.equal(expectedTimestamp);
+    });
+
+    it('sets the correct price array', async () => {
+      historicalPriceFeed = await subject();
+
+      const daysOfData = new BigNumber(1);
+      const actualPriceArray = await historicalPriceFeed.read.callAsync(daysOfData);
+
+      const expectedPriceArray = [ethPrice];
+
+      expect(JSON.stringify(actualPriceArray)).to.equal(JSON.stringify(expectedPriceArray));
+    });
+
+    describe('when the price feed is seeded with values', async () => {
+      beforeEach(async () => {
+        subjectSeededValues = [ether(160), ether(170), ether(165)];
+      });
+
+      it('should set the correct price array with 4 values', async () => {
+        historicalPriceFeed = await subject();
+
+        const daysOfData = new BigNumber(4);
+        const actualPriceArray = await historicalPriceFeed.read.callAsync(daysOfData);
```

Comment:
Why is it 4 if it's seeded with 3?

Reply:
Seeded with 3 plus the one added on deployment